src/panels/config/automation/action/types/ha-automation-action-delay.ts

```diff
@@ -20,62 +21,45 @@ export class HaDelayAction extends LitElement implements ActionElement {
 
   protected render() {
     let data: HaFormTimeData = {};
-    let templateDetected = false;
+
+    // Check for templates in action. If found, revert to YAML mode.
+    if (this.action && hasTemplate(JSON.stringify(this.action))) {
+      fireEvent(
+        this,
+        "ui-mode-not-available",
+        Error(this.hass.localize("ui.errors.config.no_template_editor_support"))
+      );
+    }
```

Comment:
Let's do this in updated, and check if the action is changed, and not in render.